src/clients/users_finish.rb

```diff
@@ -66,11 +66,20 @@ def main
       elsif @func == "Write"
         # Creating all users and their environment
 
-        # write the root password
-        UsersSimple.Write
+        if Mode.autoinst
+          # Write imported users (during autoupgrade no changes are done)
+          Users.SetWriteOnly(true)
+          @progress_orig = Progress.set(false)
+          @ret = Users.Write == ""
+          Users.SetWriteOnly(true)
```

Comment:
Just to be sure: Users.SetWriteOnly(true) called twice ? Perhaps a comment would be nice :-)